src/condor_starter.V6.1/starter.cpp

```diff
@@ -3886,8 +3892,11 @@ Starter::removeTempExecuteDir( void )
 
 #ifdef LINUX
 	if (m_lv_handle) {
-		//LVM managed... reset handle pointer to call destructor for cleanup
-		//We can't determine if the cleanup failed or not, and need to rm the working dir
+		CondorError err;
+		if ( ! m_lv_handle->CleanupLV(err)) {
+			dprintf(D_ERROR, "Failed to cleanup LV: %s\n", err.getFullText().c_str());
+			exit_code = STARTER_EXIT_IMMORTAL_LVM;
```

Comment:
should not change exit code unconditionally here, only if your new exit code is "more important"